commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/core/LinearCombinationPerformance.java

```diff
@@ -0,0 +1,383 @@
+            // Use the ill conditioned data generation method.
+            // This requires an array of at least 6.
+            final int n = Math.max(6, getLength());
+            final double[] x = new double[n];
+            final double[] y = new double[n];
+            a = new double[size][];
+            b = new double[size][];
+            // Limit precision to allow large array lengths to be generated.
+            final MathContext mathContext = new MathContext(100);
+            for (int i = 0; i < size; i++) {
+                LinearCombinationUtils.genDot(c, rng, x, y, null, mathContext);
+                a[i] = Arrays.copyOf(x, getLength());
+                b[i] = Arrays.copyOf(y, getLength());
+            }
+        }
+    }
+
+    /**
+     * The factors to multiply of a specific length.
+     */
+    @State(Scope.Benchmark)
+    public static class LengthFactors extends Factors {
+        /**
+         * The length of each factors array.
+         */
+        @Param({"2", "3", "4", "8", "16", "32", "64"})
+        private int length;
+
+        /** {@inheritDoc} */
+        @Override
+        public int getLength() {
+            return length;
+        }
+    }
+
+    /**
+     * The {@link LinearCombination} implementation.
+     */
+    @State(Scope.Benchmark)
+    public static class Calculator {
+        /**
+         * The implementation name.
+         */
+        @Param({"standard",
+                "current",
+                "dekker",
+                "dot2s",
+                "dot2", "dot3", "dot4", "dot5", "dot6", "dot7",
+                "exact",
+                "extended", "extended2", "extended_exact", "extended_exact2",
+                // Cached working double[] array.
+                // Only faster when 'length' is >16. Below this the array
+                // is small enough to be allocated locally
+                // (Search for Thread Local Allocation Buffer (TLAB))
+                "dot3c", "extendedc"})
+        private String name;
+
+        /** The 2D implementation. */
+        private TwoD twod;
+        /** The 3D implementation. */
+        private ThreeD threed;
+        /** The 4D implementation. */
+        private FourD fourd;
+        /** The ND implementation. */
+        private ND nd;
+
+        /**
+         * @return the 2D implementation
+         */
+        public TwoD getTwoD() {
+            return twod;
+        }
+
+        /**
+         * @return the 3D implementation
+         */
+        public ThreeD getThreeD() {
+            return threed;
+        }
+
+        /**
+         * @return the 4D implementation
+         */
+        public FourD getFourD() {
+            return fourd;
+        }
+
+        /**
+         * @return the ND implementation
+         */
+        public ND getND() {
+            return nd;
+        }
+
+        /**
+         * Setup the implementation.
+         */
+        @Setup
+        public void setup() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 232-250 into a separate method.